easypy/concurrency.py

```diff
@@ -377,22 +390,31 @@ def kill(wait=False):
 
 def concurrent_find(func, params, **kw):
     timeout = kw.pop("concurrent_timeout", None)
-    with async(func, list(params), **kw) as futures:
-        future = None
-        try:
-            for future in futures.as_completed(timeout=timeout):
-                if not future.exception() and future.result():
-                    futures.kill()
-                    return future.result()
-            else:
-                if future:
-                    future.result()
-        except FutureTimeoutError as exc:
-            if not timeout:
-                # ??
-                raise
-            futures.kill()
-            _logger.warning("Concurrent future timed out (%s)", exc)
+    batch_size = kw.pop("batch_size", CONCURRENT_FIND_DEFAULT_BATCH_SIZE)
```

Comment:
in case of running concurrent_find on a large collection, we don't want to launch an enormous amount of threads / greenlets.